src/element/list.rs

```diff
@@ -19,21 +17,25 @@ use std::fmt::{Display, Formatter};
 /// # }
 /// ```
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct List {
-    pub(super) children: Vec<Element>,
-}
+pub struct List(pub Sequence);
 
 impl List {
-    pub(crate) fn new(children: Vec<Element>) -> Self {
-        Self { children }
+    pub(crate) fn new(elements: impl Into<Sequence>) -> Self {
+        List(elements.into())
     }
+}
 
-    pub fn builder() -> ListBuilder {
-        ListBuilder::new()
+impl Deref for List {
+    type Target = Sequence;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
+}
```

Comment:
🗺️ This impl allows users to transparently access the underlying `Sequence`'s methods.

Reply:
[`Deref` polymorphism](https://rust-unofficial.github.io/patterns/anti_patterns/deref.html) is considered an anti-pattern. Can you explain why you went with this approach instead of any of the alternatives?

Reply:
I discussed this [in our design thread on issue 500](https://github.com/amazon-ion/ion-rust/issues/500#issuecomment-1491885057).

Reply:
Oh, I missed that. I'll go take a look.

Reply:
I just pushed a commit to replace `Deref` with `delegate!`. The result is slightly more verbose, but it's probably worth it to sidestep potential controversy.